### PR TITLE
Editor revision 2020 11 24 (review changes 2)

### PR DESCRIPTION
**Pull request by tschmidtb51**
- addresses part of changes requested in review of oasis-tcs#151
- add missing property names
- separate Acknowledgment from Acknowledgments
- add Acknowledgments to headlines
- separate Branch from Branches
- add Branches to headlines
- re-organize number to reflect change
- remove reference to branch_branches_t

### CODE REVIEW

**Comment by tschmidtb51** [comment]
Unfortunately, I forgot yesterday to create the Pull Request when providing the changes from the review in #151. 

@sthagen, @santosomar: Please review.
@sthagen: Could you please check that 0fca68b and 326ffad / 1eed0ad reflect the changes you wanted to have in there?

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 1]
```diff
@@ -464,42 +464,47 @@ Acknowledgments (`acknowledgments_t`), Branches (`branches_t`), Full Product Nam
...
+          },
+          "urls": {
             // ...
+          }
         }
       }
     },
 
-#### 3.1.1.1 Acknowledgment Type - Names
+#### 3.1.1.1 Acknowledgments Type - Acknowledgment Type
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:491]:
```suggestion
#### 3.1.1.1 Acknowledgments - Acknowledgment Type
```

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 2]
```diff
@@ -464,42 +464,47 @@ Acknowledgments (`acknowledgments_t`), Branches (`branches_t`), Full Product Nam
...
+The value type of `Acknowledgment` is object with at least 1 and at most 4 properties. Every such element acknowledges contributions by describing those that contributed.
+The properties are: `names`, `organizations`, `summary`, and `urls`.
+
+##### 3.1.1.1.1 Acknowledgments Type - Acknowledgment Type - Names
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:495]:
```suggestion
##### 3.1.1.1.1 Acknowledgments - Acknowledgment Type - Names
```

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 3]
```diff
@@ -508,46 +513,54 @@ Examples:
     Talos
     Google Project Zero
 
-#### 3.1.1.3 Acknowledgment Type - Summary
+##### 3.1.1.1.3 Acknowledgments Type - Acknowledgment Type - Summary
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:516]:
```suggestion
##### 3.1.1.1.3 Acknowledgments - Acknowledgment Type - Summary
```

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 4]
```diff
@@ -464,42 +464,47 @@ Acknowledgments (`acknowledgments_t`), Branches (`branches_t`), Full Product Nam
...
+List of acknowledged names (`names`) has value type `array` with 1 or more items holds the names of entities being recognized.
 Every such item of value type `string` with 1 or more characters represents the name of entity being recognized and contains the name of a single person. 
 
 Examples:
 
     Johann Sebastian Bach
     Albert Einstein
 
-#### 3.1.1.2 Acknowledgment Type - Organizations
+##### 3.1.1.1.2 Acknowledgments Type - Acknowledgment Type - Organizations
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:505]:
```suggestion
##### 3.1.1.1.2 Acknowledgments - Acknowledgment Type - Organizations
```

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 5]
```diff
@@ -508,46 +513,54 @@ Examples:
...
 
-Summary of the acknowledgment of value type `string` with 1 or more characters SHOULD represent any contextual details the document producers wish to make known about the acknowledgment or acknowledged parties. 
+Summary of the acknowledgment (`summary`) of value type `string` with 1 or more characters SHOULD represent any contextual details the document producers wish to make known about the acknowledgment or acknowledged parties. 
 
 Example:
 
     First analysis of Coordinated Multi-Stream Attack (CMSA)
 
-#### 3.1.1.4 Acknowledgment Type - URLs
+##### 3.1.1.1.4 Acknowledgments Type - Acknowledgment Type - URLs
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:524]:
```suggestion
##### 3.1.1.1.4 Acknowledgments - Acknowledgment Type - URLs
```

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 6]
```diff
@@ -508,46 +513,54 @@ Examples:
...
+          "product": {
+            // ...
+          }
         }
       }
     },
 
 
-#### 3.1.2.1 Branch Type - Name
+#### 3.1.2.1 Branches Type - Branch Type
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:556]:
```suggestion
#### 3.1.2.1 Branches - Branch Type
```

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 7]
```diff
@@ -508,46 +513,54 @@ Examples:
...
 
-#### 3.1.2.1 Branch Type - Name
+#### 3.1.2.1 Branches - Branch Type
+
+Every Branch holds exactly 3 properties and is a part of the hierarchical structure of the product tree. 
+The properties `name` and `type` are mandatory. In addition, the object contains either a `branches` or a `product` property. 
+
+
+
+##### 3.1.2.1.1 Branches Type - Branch Type - Name
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:563]:
```suggestion
##### 3.1.2.1.1 Branches - Branch Type - Name
```

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 8]
```diff
@@ -562,7 +575,7 @@ Examples:
     365
     PCS 7
 
-#### 3.1.2.2 Branch Type - Type
+##### 3.1.2.1.2 Branches Type - Branch Type - Type
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:578]:
```suggestion
##### 3.1.2.1.2 Branches - Branch Type - Type
```

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 9]
```diff
@@ -579,13 +592,13 @@ Valid `enum` values are:
     specification
     vendor
 
-#### 3.1.2.3 Branch Type - Branches
+##### 3.1.2.1.3 Branches Type - Branch Type - Branches
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:595]:
```suggestion
##### 3.1.2.1.3 Branches - Branch Type - Branches
```

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 10]
```diff
@@ -579,13 +592,13 @@ Valid `enum` values are:
...
 
-List of branches (`branches`) has the value type `array` providing 1 or more items of the Branch type (`branches_t`). 
+List of branches (`branches`) has the value type `branches_t`. 
 
-#### 3.1.2.4 Branch Type - Product
+##### 3.1.2.1.4 Branches Type - Branch Type - Product
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:599]:
```suggestion
##### 3.1.2.1.4 Branches - Branch Type - Product
```

**Review by sthagen — APPROVED** [review]

**Pull request merged**